### PR TITLE
MAKE-826: stream logs back to caller

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: cli/command.go [hunk 1]
```diff
@@ -110,8 +123,42 @@ func RunCMD() cli.Command {
...
+						timer := time.NewTimer(0)
+						defer timer.Stop()
+
+						for {
+							select {
+							case <-ctx.Done():
+								grip.Notice("operation canceled")
+								return
+							case <-timer.C:
+								logLines, err := client.GetLogStream(ctx, cmdID, 1000)
```
Comment (bsamek) [inline cli/command.go]: nit: make 1000 a constant

File: cli/command.go [hunk 2]
```diff
@@ -110,8 +123,42 @@ func RunCMD() cli.Command {
...
+								if err != nil {
```
Comment (bsamek) [inline cli/command.go]: Should this be resilient to a transient failure?

File: cli/utils.go [hunk 3]
```diff
@@ -248,3 +250,7 @@ func runServices(ctx context.Context, makeServices ...func(context.Context) (jas
 	<-ctx.Done()
 	return closeAllServices(closeServices)
 }
+
+func randDur(window time.Duration) time.Duration {
+	return window + time.Duration(rand.Int63n(int64(window)))
```
Comment (bsamek) [inline cli/utils.go]: This returns a number in [n, 2n), not [n/2, 3n/2). Is that deliberate?

Comment (tychoish) [reply cli/utils.go]: yeah.

File: cli/command.go (hunk 2 above)
Comment (tychoish) [reply cli/command.go]: I think the clients can/do have middleware for doing retries for network errors and the like, so it's not useful doing the retries here.

File: cli/command.go (hunk 1 above)
Comment (tychoish) [reply cli/command.go]: ok, sorta.

**Pull request merged**